hlint/hlint.dhall

```diff
@@ -35,6 +35,21 @@ in [ rule.Arguments { arguments =
    , hintNote "pure ()" "pass" "Use 'pass'"
    , hintNote "return ()" "pass" "Use 'pass'"
 
+   -- One
+   , warnSimple "(:[])" "one"
+   , warnSimple "(Data.List.NonEmpty.:|[])" "one"
+   , warnSimple "Data.Sequence.empty" "one"
+   , warnSimple "Data.Text.singleton" "one"
+   , warnSimple "Data.Text.Lazy.singleton" "one"
+   , warnSimple "Data.ByteString.singleton" "one"
+   , warnSimple "Data.ByteString.Lazy.singleton" "one"
+   , warnSimple "Data.Map.singleton" "one"
```

Comment:
People might use this function from `Data.Map.Strict` module, so I think it would be better to add one more hint. There's only one `Map` data type in the `containers` package, but `singleton` function can go from different modules.